src/main/scala/io/circe/yaml/Parser.scala

```diff
@@ -0,0 +1,133 @@
+  }
+
+  private[this] def yamlToJson(node: Node): Either[ParsingFailure, Json] = {
+    // Isn't thread-safe internally, may hence not be shared
+    val flattener: FlatteningConstructor = new FlatteningConstructor
+
+    def convertScalarNode(node: ScalarNode) = Either
+      .catchNonFatal(node.getTag match {
+        case Tag.INT if node.getValue.startsWith("0x") || node.getValue.contains("_") =>
+          Json.fromJsonNumber(flattener.construct(node) match {
+            case int: Integer         => JsonLong(int.toLong)
+            case long: java.lang.Long => JsonLong(long)
+            case bigint: java.math.BigInteger =>
+              JsonDecimal(bigint.toString)
+            case other => throw new NumberFormatException(s"Unexpected number type: ${other.getClass}")
+          })
+        case Tag.INT | Tag.FLOAT =>
+          JsonNumber.fromString(node.getValue).map(Json.fromJsonNumber).getOrElse {
+            throw new NumberFormatException(s"Invalid numeric string ${node.getValue}")
+          }
+        case Tag.BOOL =>
+          Json.fromBoolean(flattener.construct(node) match {
+            case b: java.lang.Boolean => b
+            case _                    => throw new IllegalArgumentException(s"Invalid boolean string ${node.getValue}")
+          })
+        case Tag.NULL => Json.Null
+        case CustomTag(other) =>
+          Json.fromJsonObject(JsonObject.singleton(other.stripPrefix("!"), Json.fromString(node.getValue)))
+        case other => Json.fromString(node.getValue)
+      })
+      .leftMap { err =>
+        ParsingFailure(err.getMessage, err)
+      }
+
+    def convertKeyNode(node: Node) = node match {
+      case scalar: ScalarNode => Right(scalar.getValue)
+      case _                  => Left(ParsingFailure("Only string keys can be represented in JSON", null))
+    }
+
+    if (node == null) {
+      Right(Json.False)
+    } else {
+      node match {
+        case mapping: MappingNode =>
+          flattener
+            .flatten(mapping)
+            .getValue
+            .asScala
+            .foldLeft(
+              Either.right[ParsingFailure, JsonObject](JsonObject.empty)
+            ) { (objEither, tup) =>
+              for {
+                obj <- objEither
+                key <- convertKeyNode(tup.getKeyNode)
+                value <- yamlToJson(tup.getValueNode)
+              } yield obj.add(key, value)
+            }
+            .map(Json.fromJsonObject)
+        case sequence: SequenceNode =>
+          sequence.getValue.asScala
+            .foldLeft(Either.right[ParsingFailure, List[Json]](List.empty[Json])) { (arrEither, node) =>
+              for {
+                arr <- arrEither
+                value <- yamlToJson(node)
+              } yield value :: arr
+            }
+            .map(arr => Json.fromValues(arr.reverse))
+        case scalar: ScalarNode => convertScalarNode(scalar)
+      }
+    }
+  }
+}
+
+object Parser {
+    val defaultParser = Parser()
```

Comment:
Kind of nitpicky, but can we rename this to `default` (which follows the naming convention in e.g. circe-generic-extras) and put an explicit type annotation on it?

Reply:
Yup, updated